app/models/campaign.rb

```diff
@@ -43,7 +43,7 @@ class Campaign < ActiveRecord::Base
   # Scopes #
   ##########
 
-  scope :active, -> { where('end > ?', Time.zone.now) }
+  scope :active, -> { where('end > ? OR end IS NULL', Time.zone.now) }
```

Comment:
Did you get a chance to try the ActiveRecord `.or` approach? If it works, I'd rather use that than the raw SQL.